packages/flutter_tools/templates/module/android/library_new_embedding/Flutter.tmpl/build.gradle.tmpl

```diff
@@ -30,6 +30,20 @@ group = "{{androidIdentifier}}"
 version = "1.0"
 
 android {
+    publishing {
+        singleVariant("debug") {
+            withSourcesJar()
+            withJavadocJar()
+        }
+        singleVariant("profile") {
+            withSourcesJar()
+            withJavadocJar()
+        }
+        singleVariant("release") {
+            withSourcesJar()
+            withJavadocJar()
+        }
+    }
```

Comment:
If any reviewer is knowledgeable about Maven publishing, I'd like to know if this will have unintended consequences. It looks like we were already modifying the publishing in the aar_init script as well https://github.com/flutter/flutter/blob/9ff9c67272f7f95c7d53ac3f89e0b659df9ddf3e/packages/flutter_tools/gradle/aar_init_script.gradle#L38, so it seems similar to stuff we are already doing. But I don't understand it very well

Reply:
I have some experience with maven but not with using the publishing gradle task. 
One thing I notice is that this hardcodes the variants and maybe that is ok for a template but if so we should update the documentation for adding variants to including modifying this.

Reply:
I think people actually can't modify this file - it is one of the "ephemeral" add to app gradle files that gets re-generated frequently.

Reply:
If this gets generated then we need to support custom variants.

Reply:
Resolving as this is now only relevant to follow up pr